client/my-sites/earn/memberships/index.jsx

```diff
@@ -64,14 +64,22 @@ class MembershipsSection extends Component {
 		disconnectedConnectedAccountId: null,
 	};
 	componentDidMount() {
+		this.navigateToLaunchpad();
 		this.fetchNextSubscriberPage( false, true );
 	}
 	componentDidUpdate( prevProps ) {
+		this.navigateToLaunchpad();
 		if ( prevProps.siteId !== this.props.siteId ) {
 			// Site Id changed
 			this.fetchNextSubscriberPage( false, true );
 		}
 	}
+	navigateToLaunchpad() {
+		const shouldGoToLaunchpad = this.props?.query?.stripe_connect_success === 'launchpad';
+		if ( shouldGoToLaunchpad ) {
+			window.location.assign( `/setup/newsletter/launchpad?siteSlug=${ this.props.siteSlug }` );
```

Comment:
Could this be more generic instead of `newsletter` path specifically? I'm imagining this task in other places as well.

Reply:
Good idea to make this more generalized. I have updated the url to use site_intent (which will be the same as the stepper onboarding flow).